USER_MANUAL.md

```diff
@@ -3042,6 +3042,85 @@ Using `rbreak <regex>` you can set breakpoints on multiple functions.
 -   `(gdb) break definition.kore:break -> No source file named definition.kore.`
 send `-ccopt -g` to kompile in order to generate debug info symbols.
 
+Profiling your K semantics
+--------------------------
+
+### General instructions
+<!-- _(This is this [post](https://runtimeverification.slack.com/archives/C7E701MFG/p1578937371082200), by @dwightguth, with minor changes in prose and formatting.)_ -->
+
+1. The first thing to be aware of is in order to get meaningful data, you need to build the semantics and all of its dependencies with optimizations enabled but _without the frame pointer elimination optimization_. For example, for EVM, this means rebuilding GMP, MPFR, JEMalloc, Crypto++, SECP256K1, etc with the following `exports`.
+```shell
+export CFLAGS="-g -O2 -fno-omit-frame-pointer"; export CXXFLAGS="-g -O2 -fno-omit-frame-pointer"
+```
+
+2. You can skip this step, but if you do, any samples within these libraries will not have correct stack trace information, which means you will likely not get a meaningful set of data that will tell you where the majority of time is really being spent. Don't worry about rebuilding literally every single dependency though. Just focus on the ones that you expect to take a non-negligible amount of runtime. You will be able to tell if you haven't done enough later, and you can go back and rebuild more.
+Once this is done, you then build K with optimizations and debug info enabled, like so: 
+```shell
+mvn package -Dproject.build.type="RelWithDebInfo"
+```
+
+3. Next, you build the semantics with optimizations and debug info enabled (i.e., `kompile -O3 -ccopt -g` )
```

Comment:
-O3 enables at least one optimization in the llvm backend that can cause some issues in some cases. It's probably better to recommend `kompile -ccopt -O2 --iterated`.